# Group 6 SC1015 Assignment

We will be looking into Car Features and how it will affect the MSRP of the cars. <br>

This dataset was from kaggle [Car Features and MSRP](https://www.kaggle.com/CooperUnion/cardataset) <br>

Since a description of the data was not given. Let us list out the features and their details <br>
  - __Car Make__
      * Brand of the vehicle
  - __Car Model__
      * Can also be "Name" of car
  - __Car Year__
      * When the car was manufactured
  - __Engine Fuel Type__
      * Fuel type reccomendation for the car. If it is an electric vehicle "electric" will be shown.
  - __Engine Hourse Power(HP)__
      * Refers to the power an engine produces
  - __Engine Cylinders__
      * Cylinders are located in an engine, it is  a chamber where fuel is combusted and power is generated. In this data, it refers to the amount of cylinders the Engine has.
  - __Transimission Type__
      * The way power is transmitted from an engine to the axle in the vehicle
  - __Driven Wheels__
      * Different Vehicles run on different wheel drives. Front wheel drive cars will have power transmitted to the Front 2 wheels. Rear wheel drive cars will have power transmitted to the Rear 2 wheels. All wheel drive cars will have power transmitted to all the Wheels all the time. 4WD is similar to All wheel drive cars, just with the fact that it can toggle between 4WD and 2WD. 
  - __Number of Doors__
      * Self explanatory
  - __Market Category__
      * Refers to the market audience for the cars
  - __Vehicle Size__
      * Self explanatory
  - __Vehicle Style__
      * Refers to their body style, We are familiar with cars without roofs (Convertible), normal cars (Sedans) and many more
  - __Highway MPG__
      * Refers to Miles Per Gallons, and it is used to show how far the car can travel for every gallon on an highway. (Travelling on highway does not require the car to Stop and Accelerate constantly. Therefore will usually result in Higher MPG compared to City MPG)
  - __City MPG__
      * Similar to Highway MPG, but shows how far the car travels for every gallon in the City.
  - __Popularity__
      * Data was scraped from twitter polls to indicate the popularity of the car.
  - __MSRP__
      * Refers to the manufacturer's suggested retail price
    

Find Link to Sections here <br>
   - [1. Import Libraries](#Importing)
   - [2. Data Extraction](#Data)
   - [3. Checking Data](#Check_Data)
   - [4. Performing EDA](#EDA)
  

<hr>

# 1. Import Libraries

> Importing libraries for EDA

<a id = "Importing"></a>

In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

<hr>

# 2. Data Extraction 

<a id = "Data"></a>

In [77]:
car_raw = pd.read_csv('Car.csv')

car_raw.sample(10)

Make               Model  Year                Engine Fuel Type  \
3599     Cadillac                 DTS  2009                regular unleaded   
8244   Land Rover  Range Rover Evoque  2015  premium unleaded (recommended)   
9005          GMC        Savana Cargo  2014        flex-fuel (unleaded/E85)   
2523    Chevrolet        City Express  2015                regular unleaded   
955       Porsche                 911  2016     premium unleaded (required)   
8695        Mazda                RX-7  1993                regular unleaded   
6125   Volkswagen               Jetta  2015                regular unleaded   
5796        Acura          ILX Hybrid  2014  premium unleaded (recommended)   
11222     Hyundai            Veloster  2014                regular unleaded   
8593          Kia               Rondo  2007                regular unleaded   

       Engine HP  Engine Cylinders Transmission Type      Driven_Wheels  \
3599       292.0               8.0         AUTOMATIC  front wheel drive   
8244       240.0               4.0         AUTOMATIC    all wheel drive   
9005       285.0               8.0         AUTOMATIC   rear wheel drive   
2523       131.0               4.0         AUTOMATIC  front wheel drive   
955        400.0               6.0            MANUAL    all wheel drive   
8695       255.0               NaN            MANUAL   rear wheel drive   
6125       170.0               4.0         AUTOMATIC  front wheel drive   
5796       111.0               4.0         AUTOMATIC  front wheel drive   
11222      138.0               4.0            MANUAL  front wheel drive   
8593       182.0               6.0         AUTOMATIC  front wheel drive   

       Number of Doors            Market Category Vehicle Size  Vehicle Style  \
3599               4.0                     Luxury        Large          Sedan   
8244               4.0           Crossover,Luxury      Compact        4dr SUV   
9005               3.0                  Flex Fuel        Large      Cargo Van   
2523               4.0                        NaN      Compact  Cargo Minivan   
955                2.0    Luxury,High-Performance      Compact          Coupe   
8695               2.0  Factory Tuner,Performance      Compact          Coupe   
6125               4.0                        NaN      Midsize          Sedan   
5796               4.0              Luxury,Hybrid      Compact          Sedan   
11222              3.0                  Hatchback      Compact  2dr Hatchback   
8593               4.0                        NaN      Compact          Wagon   

       highway MPG  city mpg  Popularity    MSRP  
3599            22        15        1624   54230  
8244            30        21         258   48300  
9005            17        11         549   33420  
2523            26        24        1385   21955  
955             26        18        1715  117530  
8695            23        15         586    7523  
6125            37        25         873   21995  
5796            38        39         204   28900  
11222           35        26        1439   17800  
8593            25        18        1720   20195

<hr>

# 3. Checking Data

<a id = "Check_Data"></a>

In [78]:
car_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11914 entries, 0 to 11913
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Make               11914 non-null  object 
 1   Model              11914 non-null  object 
 2   Year               11914 non-null  int64  
 3   Engine Fuel Type   11911 non-null  object 
 4   Engine HP          11845 non-null  float64
 5   Engine Cylinders   11884 non-null  float64
 6   Transmission Type  11914 non-null  object 
 7   Driven_Wheels      11914 non-null  object 
 8   Number of Doors    11908 non-null  float64
 9   Market Category    8172 non-null   object 
 10  Vehicle Size       11914 non-null  object 
 11  Vehicle Style      11914 non-null  object 
 12  highway MPG        11914 non-null  int64  
 13  city mpg           11914 non-null  int64  
 14  Popularity         11914 non-null  int64  
 15  MSRP               11914 non-null  int64  
dtypes: float64(3), int64(5

> It appears that we have 16 features and 11914 entries. However the dataset seems to be incomplete with 

> We should drop some of the columns as they would not provide sufficient data for us

> The code below shows us the columns with empty data, and the total count of empty data.

In [79]:
null_col = car_raw.columns[car_raw.isnull().any()]

car_to_clean = car_raw

car_raw[null_col].isnull().sum()

Engine Fuel Type       3
Engine HP             69
Engine Cylinders      30
Number of Doors        6
Market Category     3742
dtype: int64

>Checking Engine Fuel Type. Due to imcomplete information we will drop these columns

In [80]:
null = car_to_clean[car_to_clean.isna().any(axis=1)]
car_to_clean[car_to_clean['Engine Fuel Type'].isna()]

Make   Model  Year Engine Fuel Type  Engine HP  Engine Cylinders  \
11321  Suzuki  Verona  2004              NaN      155.0               6.0   
11322  Suzuki  Verona  2004              NaN      155.0               6.0   
11323  Suzuki  Verona  2004              NaN      155.0               6.0   

      Transmission Type      Driven_Wheels  Number of Doors Market Category  \
11321         AUTOMATIC  front wheel drive              4.0             NaN   
11322         AUTOMATIC  front wheel drive              4.0             NaN   
11323         AUTOMATIC  front wheel drive              4.0             NaN   

      Vehicle Size Vehicle Style  highway MPG  city mpg  Popularity   MSRP  
11321      Midsize         Sedan           25        17         481  17199  
11322      Midsize         Sedan           25        17         481  20199  
11323      Midsize         Sedan           25        17         481  18499

In [81]:
car_to_clean.iloc[11324:11327]

Make   Model  Year  Engine Fuel Type  Engine HP  Engine Cylinders  \
11324  Suzuki  Verona  2005  regular unleaded      155.0               6.0   
11325  Suzuki  Verona  2005  regular unleaded      155.0               6.0   
11326  Suzuki  Verona  2005  regular unleaded      155.0               6.0   

      Transmission Type      Driven_Wheels  Number of Doors Market Category  \
11324         AUTOMATIC  front wheel drive              4.0             NaN   
11325         AUTOMATIC  front wheel drive              4.0             NaN   
11326         AUTOMATIC  front wheel drive              4.0             NaN   

      Vehicle Size Vehicle Style  highway MPG  city mpg  Popularity   MSRP  
11324      Midsize         Sedan           25        18         481  19349  
11325      Midsize         Sedan           25        18         481  21049  
11326      Midsize         Sedan           25        18         481  17549

Looking at the next 3 cars. They are of similar make and model, just different years. Therefore we will let the 2004 model be regular unleaded.

In [82]:
car_to_clean_map = {label:idx for idx,label in enumerate(car_to_clean["Engine Fuel Type"].unique())}
car_to_clean["Engine Fuel Type"] = car_to_clean["Engine Fuel Type"].map(car_to_clean_map)

car_to_clean.at[11321, 'Engine Fuel Type'] = 1
car_to_clean.at[11322, 'Engine Fuel Type'] = 1
car_to_clean.at[11323, 'Engine Fuel Type'] = 1

car_to_clean.iloc[11321:11324]

Make   Model  Year  Engine Fuel Type  Engine HP  Engine Cylinders  \
11321  Suzuki  Verona  2004                 1      155.0               6.0   
11322  Suzuki  Verona  2004                 1      155.0               6.0   
11323  Suzuki  Verona  2004                 1      155.0               6.0   

      Transmission Type      Driven_Wheels  Number of Doors Market Category  \
11321         AUTOMATIC  front wheel drive              4.0             NaN   
11322         AUTOMATIC  front wheel drive              4.0             NaN   
11323         AUTOMATIC  front wheel drive              4.0             NaN   

      Vehicle Size Vehicle Style  highway MPG  city mpg  Popularity   MSRP  
11321      Midsize         Sedan           25        17         481  17199  
11322      Midsize         Sedan           25        17         481  20199  
11323      Midsize         Sedan           25        17         481  18499

>Checking Engine HP we can see that most electric cars do not have their HP allocated. Therefore we will use the mean to input the data

In [83]:
car_to_clean[car_to_clean['Engine HP'].isna()]

Make        Model  Year  Engine Fuel Type  Engine HP  \
539      FIAT         500e  2015                 5        NaN   
540      FIAT         500e  2016                 5        NaN   
541      FIAT         500e  2017                 5        NaN   
2905  Lincoln  Continental  2017                 2        NaN   
2906  Lincoln  Continental  2017                 2        NaN   
...       ...          ...   ...               ...        ...   
9850      Kia      Soul EV  2015                 5        NaN   
9851      Kia      Soul EV  2015                 5        NaN   
9852      Kia      Soul EV  2016                 5        NaN   
9853      Kia      Soul EV  2016                 5        NaN   
9854      Kia      Soul EV  2016                 5        NaN   

      Engine Cylinders Transmission Type      Driven_Wheels  Number of Doors  \
539                0.0      DIRECT_DRIVE  front wheel drive              2.0   
540                0.0      DIRECT_DRIVE  front wheel drive              2.0   
541                0.0      DIRECT_DRIVE  front wheel drive              2.0   
2905               6.0         AUTOMATIC    all wheel drive              4.0   
2906               6.0         AUTOMATIC  front wheel drive              4.0   
...                ...               ...                ...              ...   
9850               0.0      DIRECT_DRIVE  front wheel drive              4.0   
9851               0.0      DIRECT_DRIVE  front wheel drive              4.0   
9852               0.0      DIRECT_DRIVE  front wheel drive              4.0   
9853               0.0      DIRECT_DRIVE  front wheel drive              4.0   
9854               0.0      DIRECT_DRIVE  front wheel drive              4.0   

     Market Category Vehicle Size  Vehicle Style  highway MPG  city mpg  \
539        Hatchback      Compact  2dr Hatchback          108       122   
540        Hatchback      Compact  2dr Hatchback          103       121   
541        Hatchback      Compact  2dr Hatchback          103       121   
2905          Luxury        Large          Sedan           25        17   
2906          Luxury        Large          Sedan           27        18   
...              ...          ...            ...          ...       ...   
9850             NaN      Compact          Wagon           92       120   
9851             NaN      Compact          Wagon           92       120   
9852             NaN      Compact          Wagon           92       120   
9853             NaN      Compact          Wagon           92       120   
9854             NaN      Compact          Wagon           92       120   

      Popularity   MSRP  
539          819  31800  
540          819  31800  
541          819  31800  
2905          61  55915  
2906          61  62915  
...          ...    ...  
9850        1720  35700  
9851        1720  33700  
9852        1720  33950  
9853        1720  31950  
9854        1720  35950  

[69 rows x 16 columns]

In [84]:
car_to_clean_HP = np.where(car_to_clean['Engine HP'].isnull())

for i in range(len(car_to_clean_HP[0])):
    car_to_clean.loc[car_to_clean_HP[0][i], 'Engine HP'] = np.mean(car_to_clean['Engine HP'])
    
car_to_clean.iloc[540:542]

Make Model  Year  Engine Fuel Type  Engine HP  Engine Cylinders  \
540  FIAT  500e  2016                 5  249.38607               0.0   
541  FIAT  500e  2017                 5  249.38607               0.0   

    Transmission Type      Driven_Wheels  Number of Doors Market Category  \
540      DIRECT_DRIVE  front wheel drive              2.0       Hatchback   
541      DIRECT_DRIVE  front wheel drive              2.0       Hatchback   

    Vehicle Size  Vehicle Style  highway MPG  city mpg  Popularity   MSRP  
540      Compact  2dr Hatchback          103       121         819  31800  
541      Compact  2dr Hatchback          103       121         819  31800

>Checking Engine Cylinders we can see that all electric cars are missing values. This is due to electric vehicles not having engines.

>Another interesting vehicle model that suffers from this is the RX7 and RX8. This is due to the vehicles having rotary engines that do not have cylinders.

> Therefore we will just let the cars Engine Cylinders = 0

In [85]:
car_to_clean_cylin = np.where(car_to_clean['Engine Cylinders'].isna())
car_to_clean[car_to_clean['Engine Cylinders'].isna()]

Make    Model  Year  Engine Fuel Type  Engine HP  \
1983   Chevrolet  Bolt EV  2017                 5  200.00000   
1984   Chevrolet  Bolt EV  2017                 5  200.00000   
3716  Volkswagen   e-Golf  2015                 5  115.00000   
3717  Volkswagen   e-Golf  2015                 5  115.00000   
3718  Volkswagen   e-Golf  2016                 5  115.00000   
3719  Volkswagen   e-Golf  2016                 5  115.00000   
5778  Mitsubishi   i-MiEV  2014                 5  249.38607   
5779  Mitsubishi   i-MiEV  2016                 5   66.00000   
5780  Mitsubishi   i-MiEV  2017                 5   66.00000   
8373      Toyota  RAV4 EV  2012                 5  154.00000   
8695       Mazda     RX-7  1993                 1  255.00000   
8696       Mazda     RX-7  1994                 1  255.00000   
8697       Mazda     RX-7  1995                 1  255.00000   
8698       Mazda     RX-8  2009                 0  232.00000   
8699       Mazda     RX-8  2009                 0  212.00000   
8700       Mazda     RX-8  2009                 0  232.00000   
8701       Mazda     RX-8  2009                 0  232.00000   
8702       Mazda     RX-8  2009                 0  232.00000   
8703       Mazda     RX-8  2009                 0  212.00000   
8704       Mazda     RX-8  2009                 0  212.00000   
8705       Mazda     RX-8  2010                 0  232.00000   
8706       Mazda     RX-8  2010                 0  212.00000   
8707       Mazda     RX-8  2010                 0  212.00000   
8708       Mazda     RX-8  2010                 0  232.00000   
8709       Mazda     RX-8  2010                 0  232.00000   
8710       Mazda     RX-8  2011                 0  212.00000   
8711       Mazda     RX-8  2011                 0  232.00000   
8712       Mazda     RX-8  2011                 0  232.00000   
8713       Mazda     RX-8  2011                 0  232.00000   
8714       Mazda     RX-8  2011                 0  212.00000   

      Engine Cylinders Transmission Type      Driven_Wheels  Number of Doors  \
1983               NaN      DIRECT_DRIVE  front wheel drive              4.0   
1984               NaN      DIRECT_DRIVE  front wheel drive              4.0   
3716               NaN      DIRECT_DRIVE  front wheel drive              4.0   
3717               NaN      DIRECT_DRIVE  front wheel drive              4.0   
3718               NaN      DIRECT_DRIVE  front wheel drive              4.0   
3719               NaN      DIRECT_DRIVE  front wheel drive              4.0   
5778               NaN      DIRECT_DRIVE   rear wheel drive              4.0   
5779               NaN      DIRECT_DRIVE   rear wheel drive              4.0   
5780               NaN      DIRECT_DRIVE   rear wheel drive              4.0   
8373               NaN      DIRECT_DRIVE  front wheel drive              4.0   
8695               NaN            MANUAL   rear wheel drive              2.0   
8696               NaN            MANUAL   rear wheel drive              2.0   
8697               NaN            MANUAL   rear wheel drive              2.0   
8698               NaN            MANUAL   rear wheel drive              4.0   
8699               NaN         AUTOMATIC   rear wheel drive              4.0   
8700               NaN            MANUAL   rear wheel drive              4.0   
8701               NaN            MANUAL   rear wheel drive              4.0   
8702               NaN            MANUAL   rear wheel drive              4.0   
8703               NaN         AUTOMATIC   rear wheel drive              4.0   
8704               NaN         AUTOMATIC   rear wheel drive              4.0   
8705               NaN            MANUAL   rear wheel drive              4.0   
8706               NaN         AUTOMATIC   rear wheel drive              4.0   
8707               NaN         AUTOMATIC   rear wheel drive              4.0   
8708               NaN            MANUAL   rear wheel drive              4.0   
8709               NaN     

In [86]:
car_to_clean.loc[car_to_clean_cylin[0][0:30], 'Engine Cylinders'] = 0
car_to_clean[car_to_clean['Engine Cylinders'].isna()]

Empty DataFrame
Columns: [Make, Model, Year, Engine Fuel Type, Engine HP, Engine Cylinders, Transmission Type, Driven_Wheels, Number of Doors, Market Category, Vehicle Size, Vehicle Style, highway MPG, city mpg, Popularity, MSRP]
Index: []

>Checking the missing values. Due to the small sample size of this column we will manually check for the information online and input them into the dataframe

>[Ferrari FF](https://en.wikipedia.org/wiki/Ferrari_FF) __Has 3 doors__

>[Tesla Model S](https://en.wikipedia.org/wiki/Tesla_Model_S) __Has 4 doors__

In [87]:
car_to_clean[car_to_clean['Number of Doors'].isna()]

Make    Model  Year  Engine Fuel Type  Engine HP  Engine Cylinders  \
4666  Ferrari       FF  2013                 0  651.00000              12.0   
6930    Tesla  Model S  2016                 5  249.38607               0.0   
6931    Tesla  Model S  2016                 5  249.38607               0.0   
6932    Tesla  Model S  2016                 5  249.38607               0.0   
6933    Tesla  Model S  2016                 5  249.38607               0.0   
6934    Tesla  Model S  2016                 5  249.38607               0.0   

     Transmission Type     Driven_Wheels  Number of Doors  \
4666  AUTOMATED_MANUAL   all wheel drive              NaN   
6930      DIRECT_DRIVE   all wheel drive              NaN   
6931      DIRECT_DRIVE   all wheel drive              NaN   
6932      DIRECT_DRIVE   all wheel drive              NaN   
6933      DIRECT_DRIVE  rear wheel drive              NaN   
6934      DIRECT_DRIVE   all wheel drive              NaN   

              Market Category Vehicle Size Vehicle Style  highway MPG  \
4666  Exotic,High-Performance        Large         Coupe           16   
6930       Exotic,Performance        Large         Sedan          105   
6931       Exotic,Performance        Large         Sedan          101   
6932  Exotic,High-Performance        Large         Sedan          105   
6933       Exotic,Performance        Large         Sedan          100   
6934       Exotic,Performance        Large         Sedan          107   

      city mpg  Popularity    MSRP  
4666        11        2774  295000  
6930       102        1391   79500  
6931        98        1391   66000  
6932        92        1391  134500  
6933        97        1391   74500  
6934       101        1391   71000

In [88]:
car_to_clean.loc[4666, 'Number of Doors'] = 3
car_to_clean.loc[6930, 'Number of Doors'] = 4
car_to_clean.loc[6931, 'Number of Doors'] = 4
car_to_clean.loc[6932, 'Number of Doors'] = 4
car_to_clean.loc[6933, 'Number of Doors'] = 4
car_to_clean.loc[6934, 'Number of Doors'] = 4

> Due to the sheer size of the missing values in the "Market Category" columns we will categorize the missing values as No Cat

In [89]:
car_to_clean_mark = np.where(car_to_clean['Market Category'].isna())
car_to_clean.loc[car_to_clean_mark[0], 'Market Category'] = 'No Cat' 

In [90]:
car_to_clean['Transmission Type'].value_counts()

AUTOMATIC           8266
MANUAL              2935
AUTOMATED_MANUAL     626
DIRECT_DRIVE          68
UNKNOWN               19
Name: Transmission Type, dtype: int64

In [91]:
car_to_clean.loc[car_to_clean['Transmission Type'] == 'AUTOMATIC', 'Transmission Type'] = 1
car_to_clean.loc[car_to_clean['Transmission Type'] == 'MANUAL', 'Transmission Type'] = 2
car_to_clean.loc[car_to_clean['Transmission Type'] == 'AUTOMATED_MANUAL', 'Transmission Type'] = 3
car_to_clean.loc[car_to_clean['Transmission Type'] == 'DIRECT_DRIVE', 'Transmission Type'] = 4
car_to_clean.loc[car_to_clean['Transmission Type'] == 'UNKNOWN', 'Transmission Type'] = 5

In [73]:


# apart from auto and manual there were other values so I numbered them accordingly.
# can merge automated manual and direct drive accordingly if u guys want me to?
# also what do we do for unknown? 19 counts so technically can be fixed manually by checking for values online?
# lemme know what to do for this.



In [92]:
car_to_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11914 entries, 0 to 11913
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Make               11914 non-null  object 
 1   Model              11914 non-null  object 
 2   Year               11914 non-null  int64  
 3   Engine Fuel Type   11914 non-null  int64  
 4   Engine HP          11914 non-null  float64
 5   Engine Cylinders   11914 non-null  float64
 6   Transmission Type  11914 non-null  object 
 7   Driven_Wheels      11914 non-null  object 
 8   Number of Doors    11914 non-null  float64
 9   Market Category    11914 non-null  object 
 10  Vehicle Size       11914 non-null  object 
 11  Vehicle Style      11914 non-null  object 
 12  highway MPG        11914 non-null  int64  
 13  city mpg           11914 non-null  int64  
 14  Popularity         11914 non-null  int64  
 15  MSRP               11914 non-null  int64  
dtypes: float64(3), int64(6

In [93]:
car_to_clean['Transmission Type'] = car_to_clean['Transmission Type'].astype('int64')

In [49]:
car_to_clean['Market Category'].value_counts()

No Cat                                             3742
Crossover                                          1110
Flex Fuel                                           872
Luxury                                              855
Luxury,Performance                                  673
                                                   ... 
Performance,Hybrid                                    1
Exotic,Luxury,High-Performance,Hybrid                 1
Crossover,Exotic,Luxury,Performance                   1
Crossover,Exotic,Luxury,High-Performance              1
Flex Fuel,Factory Tuner,Luxury,High-Performance       1
Name: Market Category, Length: 72, dtype: int64

In [57]:


# converted the Transmission type to int, others left them as object. do we need to change those to category maybe?
# maybe we split the 'market category' column into multiple columns like 'MC_crossover', 'MC_luxury', 'MC_Performance' etc.
# and make the values 1 or 0 depending on whether it previously had the corresponding value?
# What do you guys think?



In [94]:


# not sure what do you mean by if a car has renewed its COE... how do I check this?



In [96]:
def age(car_to_clean):
    val = 2022 - car_to_clean['Year']
    return val

car_to_clean['Age'] = car_to_clean.apply(age, axis = 1)

In [99]:
car_to_clean['Make'].value_counts()

Chevrolet        1123
Ford              881
Volkswagen        809
Toyota            746
Dodge             626
Nissan            558
GMC               515
Honda             449
Mazda             423
Cadillac          397
Mercedes-Benz     353
Suzuki            351
BMW               334
Infiniti          330
Audi              328
Hyundai           303
Volvo             281
Subaru            256
Acura             252
Kia               231
Mitsubishi        213
Lexus             202
Buick             196
Chrysler          187
Pontiac           186
Lincoln           164
Oldsmobile        150
Land Rover        143
Porsche           136
Saab              111
Aston Martin       93
Plymouth           82
Bentley            74
Ferrari            69
FIAT               62
Scion              60
Maserati           58
Lamborghini        52
Rolls-Royce        31
Lotus              29
Tesla              18
HUMMER             17
Maybach            16
Alfa Romeo          5
McLaren             5
Bugatti   

In [100]:
car_to_clean

Make       Model  Year  Engine Fuel Type  Engine HP  \
0          BMW  1 Series M  2011                 0      335.0   
1          BMW    1 Series  2011                 0      300.0   
2          BMW    1 Series  2011                 0      300.0   
3          BMW    1 Series  2011                 0      230.0   
4          BMW    1 Series  2011                 0      230.0   
...        ...         ...   ...               ...        ...   
11909    Acura         ZDX  2012                 0      300.0   
11910    Acura         ZDX  2012                 0      300.0   
11911    Acura         ZDX  2012                 0      300.0   
11912    Acura         ZDX  2013                 2      300.0   
11913  Lincoln      Zephyr  2006                 1      221.0   

       Engine Cylinders  Transmission Type      Driven_Wheels  \
0                   6.0                  2   rear wheel drive   
1                   6.0                  2   rear wheel drive   
2                   6.0                  2   rear wheel drive   
3                   6.0                  2   rear wheel drive   
4                   6.0                  2   rear wheel drive   
...                 ...                ...                ...   
11909               6.0                  1    all wheel drive   
11910               6.0                  1    all wheel drive   
11911               6.0                  1    all wheel drive   
11912               6.0                  1    all wheel drive   
11913               6.0                  1  front wheel drive   

       Number of Doors                        Market Category Vehicle Size  \
0                  2.0  Factory Tuner,Luxury,High-Performance      Compact   
1                  2.0                     Luxury,Performance      Compact   
2                  2.0                Luxury,High-Performance      Compact   
3                  2.0                     Luxury,Performance      Compact   
4                  2.0                                 Luxury      Compact   
...                ...                                    ...          ...   
11909              4.0             Crossover,Hatchback,Luxury      Midsize   
11910              4.0             Crossover,Hatchback,Luxury      Midsize   
11911              4.0             Crossover,Hatchback,Luxury      Midsize   
11912              4.0             Crossover,Hatchback,Luxury      Midsize   
11913              4.0                                 Luxury      Midsize   

       Vehicle Style  highway MPG  city mpg  Popularity   MSRP  Age  
0              Coupe           26        19        3916  46135   11  
1        Convertible           28        19        3916  40650   11  
2              Coupe           28        20        3916  36350   11  
3              Coupe           28        18        3916  29450   11  
4        Convertible           28        18        3916  34500   11  
...              ...          ...       ...         ...    ...  ...  
11909  4dr Hatchback           23        16         204  46120   10  
11910  4dr Hatchback           23        16         204  56670   10  
11911  4dr Hatchback           23        16         204  50620   10  
11912  4dr Hatchback           23        16         204  50920    9  
11913          Sedan           26        17          61  28995   16  

[11914 rows x 17 columns]